### PR TITLE
tboot: upgrade to 1.9.12

### PR DESCRIPTION
**Pull request by eric-ch**
Patches/changes already upstream or no longer necessary:
0001-config-Allow-build-system-integration.patch
0014-safestringlib-Attend-GCC-warnings.patch
gcc9.patch

Backports:
0001-Do-not-try-to-read-EFI-mem-map-when-booted-with-mult.patch

Forward sealing tested on two Dell latitude E7470 (TPM2.0 and TPM1.2) both UEFI and legacy
- TPM 1.2 on UEFI forward sealing requires: https://github.com/OpenXT/xenclient-oe/pull/1412
- All other tests succeeded sealing using master, forward sealing to a build with TBoot 1.9.12.

### CODE REVIEW

**Comment by jandryuk** [comment]
Patch 0006-pcr-calc-Add-pcr-calculator-tool.patch is a mess of a diff.  The in-tree 1.9.9 version in unsorted, and the diffstat has not been updated as it changed with the addition of acmmatch and other stuff.  Manually re-arranging hunks, vimdiff shows the code is unchanged.

One thing I noticed is  this enum hasn't been updated for 1.9.12:
+ typedef enum {
+       TB_196,
+       TB_199,
+} tb_version_t;

Is that okay?

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/44): [$master-custom-44](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom/44/)

**Comment by eric-ch** [comment]
@jandryuk This is not okay at all, I checked my test-build and the change to `/etc/openxt/seal-system.conf` was not included in it. Which means TPM 2.0 (the ones using the pcr-calc event emulator that requires this option) will fail with `invalid tboot version: 1.9.12.` when forward sealing.

**Comment by eric-ch** [comment]
#### v2:

- Refactor pcr-calc to handle TBoot version without requiring source changes.

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/45): [$master-custom-45](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom/45/)

File: recipes-security/tboot/tboot-1.9.12/0006-pcr-calc-Add-pcr-calculator-tool.patch [hunk 1]
```diff
@@ -1,26 +1,76 @@
...
+module_hash [-hjz] [-e hash_str] -c cmd_line module_file
+ -h Help: will print out this help message.
+ -j module is compressed with BZip2
+ -z module is compressed with GZip
+ -e hash_str Extend Hash: has to be extended with module hash.
+ -c cmd_line Command Line: the command line passed.
+ -a hash algorithm to use.
+ module_file: file name of module to hash.
+
+module_hash [-hjz] [-e hash_str] -c cmd_line module_file
```
Comment (jandryuk) [inline recipes-security/tboot/tboot-1.9.12/0006-pcr-calc-Add-pcr-calculator-tool.patch:18]: module_hash is listed twice.

**Comment by jandryuk** [comment]
I tested locally and successful forward sealed tboot 1.9.9 -> 1.9.12 on UEFI/TPM1.2.

**Comment by crogers1** [comment]
@jandryuk are you still +1 on this PR?

**Comment by jandryuk** [comment]
I think I wanted to test tboot 1.9.9->1.9.12 forward seal on UEFI TPM 2.0.  I can't remember precisely why now... https://github.com/OpenXT/xenclient-oe/pull/1412 fixed TPM 1.2 and I tested that as written above.  I don't have a TPM 2.0 machine readily available for testing this which is part of the hold up.  Has anyone else tested it?

**Comment by crogers1** [comment]
Not that I'm aware of.  I'll do another build on latest master and try it out on one of my UEFI tpm 2.0 boxes

**Comment by crogers1** [comment]
[build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/56)

**Comment by crogers1** [comment]
tested master-custom 56. upgrade from tboot 1.9.9 to 1.9.12 on TPM 2.0 UEFI forward sealed succeeded.  Also worked with fresh install.

Merging later.

**Comment by jandryuk** [comment]
Thanks for testing UEFI TPM2.0.

**Pull request merged**